services/eventservice/src/data_manager.rs

```diff
@@ -0,0 +1,301 @@
+        match self.db.generate_id() {
+            Ok(id) => Ok(id),
+            Err(e) => Err(Box::new(e)),
+        }
+    }
+
+    pub fn create_event(&self, new_event: &Event) -> Result<(), Box<dyn std::error::Error>> {
+        let id = new_event.id.to_be_bytes().clone();
+        let org_id = new_event.organizer.to_be_bytes().clone();
+        let cat = new_event.category.as_bytes();
+        let events = self.db.open_tree(&self.events_name)?;
+        events.insert(id.clone(), new_event.to_json()?.as_bytes())?;
+        let organizers = self.db.open_tree(&self.organizers_name)?;
+        let categories = self.db.open_tree(&self.categories_name)?;
+        if organizers.contains_key(&org_id)? {
+            organizers.merge(&org_id, id)?;
+        } else {
+            organizers.insert(org_id, id.to_vec())?;
+        }
+        if categories.contains_key(&cat)? {
+            categories.merge(&cat, id)?;
+        } else {
+            return Err(Box::new(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                "Unknown category",
+            )));
+        }
+        Ok(())
+    }
+
+    pub fn delete_event(&self, event_id: &u64) -> Result<(), Box<dyn std::error::Error>> {
+        let id = event_id.to_be_bytes();
+        let events = self.db.open_tree(&self.events_name)?;
+        if !events.contains_key(&id)? {
+            return Ok(());
+        }
+        let event = Event::from_json(&String::from_utf8(events.get(id)?.unwrap().to_vec())?)?;
+        let org_id = event.organizer.to_be_bytes().clone();
+        let cat = event.category.as_bytes();
+        events.remove(id)?;
+        let organizers = self.db.open_tree(&self.organizers_name)?;
+        let categories = self.db.open_tree(&self.categories_name)?;
+        if organizers.contains_key(&org_id)? {
+            let ids = organizers.get(&org_id)?.unwrap();
+            let new_ids = EventManager::_remove_id(&ids[..], &id);
+            if new_ids.is_some() {
+                organizers.compare_and_swap(org_id, Some(ids), new_ids)??;
+            }
+        }
+        if categories.contains_key(&cat)? {
+            let ids = categories.get(&cat)?.unwrap();
+            let new_ids = EventManager::_remove_id(&ids[..], &id);
+            if new_ids.is_some() {
+                categories.compare_and_swap(cat, Some(ids), new_ids)??;
+            }
+        }
+        Ok(())
+    }
+
+    pub fn update_event(&self, new_event: &Event) -> Result<(), Box<dyn std::error::Error>> {
+        let id = new_event.id.to_be_bytes().clone();
+        let events = self.db.open_tree(&self.events_name)?;
+        events.insert(id, new_event.to_json().unwrap().as_bytes())?;
+        Ok(())
+    }
+
+    pub fn get_event(&self, event_id: &u64) -> Result<Event, Box<dyn std::error::Error>> {
+        let id = event_id.to_be_bytes();
+        let events = self.db.open_tree(&self.events_name)?;
+        let event = events.get(id)?;
+        if event.is_none() {
```

Comment:
Please, use `match` expression as you do in other such methods in the file

Reply:
Done